extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java

```diff
@@ -136,6 +144,37 @@ static String toMimeType(String flickrFormat) {
     }
   }
 
+  private ExportResult<PhotosContainerResource> exportPhotosContainer(
+      PhotosContainerResource container) {
+    ImmutableList.Builder<PhotoAlbum> albumBuilder = ImmutableList.builder();
+    ImmutableList.Builder<PhotoModel> photosBuilder = ImmutableList.builder();
+    List<IdOnlyContainerResource> subResources = new ArrayList<>();
+
+    try {
+      for (PhotoAlbum album : container.getAlbums()) {
+        Photoset photoset = photosetsInterface.getInfo(album.getId());
+        // Saving data to the album allows the target service to recreate the album structure
+        albumBuilder.add(
+            new PhotoAlbum(photoset.getId(), photoset.getTitle(), photoset.getDescription()));
+        // Adding subresources tells the framework to recall export to get all the photos
+        subResources.add(new IdOnlyContainerResource(photoset.getId()));
+      }
+
+      for (PhotoModel photo : container.getPhotos()) {
+        Photo p = photosInterface.getInfo(photo.getDataId(), null);
+        photosBuilder.add(toCommonPhoto(p, null));
+      }
+    } catch (FlickrException e) {
+      return new ExportResult<>(e);
+    }
+
+    PhotosContainerResource photosContainerResource =
+        new PhotosContainerResource(albumBuilder.build(), photosBuilder.build());
+    ContinuationData continuationData = new ContinuationData(null);
+    subResources.forEach(resource -> continuationData.addContainerResource(resource));
+    return new ExportResult<>(ResultType.CONTINUE, photosContainerResource, continuationData);
```

Comment:
What would happen if there are no albums and no photos and each of these is empty?

Reply:
Then there would be nothing to import. The job will be counted as "success". 

I guess what we do here depends on what the expectation is if there is no content to export. If they specified an invalid photo/album id the lookup fails and an error gets thrown. 

If we want the job to fail, we can add a precondition check, but again not sure if thats what we want to do.